tool/src/demo_gen/terminus.rs

```diff
@@ -0,0 +1,380 @@
+        let mut root =
+            MethodDependency::new(self.get_constructor_js(type_name.to_string(), method));
+
+        // And then we just treat the terminus as a regular constructor method:
+        self.terminus_info.node_call_stack = self.evaluate_constructor(method, &mut root);
+
+        let type_n = type_name.clone();
+        let format = self
+            .formatter
+            .fmt_import_statement(&type_n, false, "./js/".into());
+
+        self.terminus_info.imports.insert(format);
+    }
+
+    /// Currently unused, plan to hopefully use this in the future for quickly grabbing parameter information.
+    fn _get_type_demo_attrs(&self, ty: &Type) -> Option<DemoInfo> {
+        ty.id()
+            .map(|id| self.tcx.resolve_type(id).attrs().demo_attrs.clone())
+    }
+
+    /// Take a parameter passed to a terminus (or a constructor), and either:
+    /// 1. Add it to the list of parameters that the terminus function takes for the render engine to call.
+    /// 2. Go a step deeper and look at its possible constructors to call evaluate_param on.
+    ///
+    /// `node` - Represents the current function of the parameter we're evaluating. See [`MethodDependency`] for more on its purpose.
+    fn evaluate_param(
+        &mut self,
+        param_type: &Type,
+        param_name: String,
+        node: &mut MethodDependency,
+        method_attrs: DemoInfo,
+    ) {
+        let attrs = Some(method_attrs); /*self.get_type_demo_attrs(param_type).or(Some(method_attrs));*/
+        // Helper function for quickly passing a parameter to both our node and the render terminus.
+        let out_param = |type_name| {
+            // This only works for enums, since otherwise we break the type into its component parts.
+            let label = attrs
+                .and_then(|attrs| {
+                    let label = attrs
+                        .input_cfg
+                        .get(&param_name)
+                        .map(|cfg| cfg.label.clone())
+                        .unwrap_or_default();
+
+                    if label.is_empty() {
+                        None
+                    } else {
+                        Some(label)
+                    }
+                })
+                .unwrap_or(heck::AsUpperCamelCase(param_name.clone()).to_string());
+
+            let mut param_info = ParamInfo {
+                js: param_name,
+                label,
+                type_name,
+            };
+
+            self.terminus_info.params.push(param_info.clone());
+
+            // Grab arguments without having to name them
+            param_info.js = format!("terminusArgs[{}]", self.terminus_info.params.len() - 1);
+            node.params.push(param_info);
+        };
+
+        // TODO: I think we need to check for struct and opaque types as to whether or not these have attributes that label them as provided as a parameter.
+        match param_type {
+            Type::Primitive(_) | Type::Enum(_) | Type::Slice(_) => {
```

Comment:
nit: since we have `out_param` already I'd just split these into three match statements, matching again on the same thing seems silly